networks/suzuka/suzuka-client/src/load_soak_testing/mod.rs

```diff
@@ -0,0 +1,394 @@
+				}
+			};
+			let metrics_scenario = serde_json::to_string(&metrics)
+				.unwrap_or("Metric serialization error.".to_string());
+			tracing::info!(target:EXEC_LOG_FILTER, metrics_scenario);
+			scenario_results.push(metrics);
+		}
+		scenario_results
+	}
+
+	async fn soak_runner_in_a_loop(
+		self,
+		create_scanario: Arc<scenario::CreateScenarioFn>,
+		duration: std::time::Duration,
+	) -> Vec<ScenarioExecMetric> {
+		let initial_start_time = std::time::Instant::now();
+
+		let mut set = tokio::task::JoinSet::new();
+		//start min scenario
+		self.scenario_chunk.into_iter().for_each(|id| {
+			let create_scanario = create_scanario.clone();
+			set.spawn(futures::future::join(
+				futures::future::ready(id),
+				run_scenarion_in_loop(id, create_scanario, duration.clone()),
+			));
+		});
+
+		let mut scenario_results = vec![];
+		while let Some(res) = set.join_next().await {
+			let metrics = match res {
+				Ok((id, Ok(elapse))) => ScenarioExecMetric::new_ok(id, elapse),
+				Ok((id, Err(err))) => {
+					let log = format!("Scenario:{id} execution failed because: {err}");
+					tracing::info!(target:EXEC_LOG_FILTER, log);
+					tracing::warn!(log);
+					let elapse = initial_start_time.elapsed().as_millis();
+					ScenarioExecMetric::new_err(id, elapse)
+				}
+				Err(err) => {
+					tracing::warn!("Error during scenario spawning: {err}");
+					let elapse = initial_start_time.elapsed().as_millis();
+					ScenarioExecMetric::new_err(0, elapse)
+				}
+			};
+			let metrics_scenario = serde_json::to_string(&metrics)
+				.unwrap_or("Metric serialization error.".to_string());
+			tracing::info!(target:EXEC_LOG_FILTER, metrics_scenario);
+			scenario_results.push(metrics);
+		}
+		scenario_results
+	}
+}
+
+async fn run_scenarion_in_loop(
+	id: usize,
+	create_scanario: Arc<scenario::CreateScenarioFn>,
+	duration: Duration,
+) -> anyhow::Result<u128> {
+	let start_time = std::time::Instant::now();
+	let mut average_time = 0;
+	loop {
+		let elapse = start_time.elapsed();
+		if elapse > duration {
+			break;
+		}
+
+		tracing::info!("{id} start new test");
+		let exec_start_time = std::time::Instant::now();
+		let scenario = create_scanario(id);
+		scenario.run().await?;
+		let exec_elapse = exec_start_time.elapsed().as_millis();
+		if average_time == 0 {
+			average_time = exec_elapse;
+		} else {
+			average_time = (exec_elapse + average_time) / 2;
+		}
+		tracing::info!("{id} end test exec_elapse:{exec_elapse} average_time:{average_time}");
+	}
+	Ok(average_time)
+}
+
+#[derive(Serialize, Deserialize)]
+struct ScenarioExecMetric {
+	scenario_id: usize,
+	elapse_millli: u128,
+	result: ScenarioExecResult,
+}
+
+impl ScenarioExecMetric {
+	fn new_ok(scenario_id: usize, elapse_millli: u128) -> Self {
+		ScenarioExecMetric { scenario_id, elapse_millli, result: ScenarioExecResult::Ok }
+	}
+	fn new_err(scenario_id: usize, elapse_millli: u128) -> Self {
+		ScenarioExecMetric { scenario_id, elapse_millli, result: ScenarioExecResult::Fail }
```

Comment:
I can't immedietly see why we would need a `new_ok` and a `new_err` method if a `new` would just return `Self` where ofcourse result is a `ScenarioExecResult`.

Reply:
Yeah, this seems a bit of an anti-pattern here.